File: algorithms/400-499/413. Arithmetic Slices.py
class Solution:
    def numberOfArithmeticSlices(self, A: List[int]) -> int:
        if not A:
            return 0
        n = len(A)
        if n <= 2:
            return 0
        dp = [0 for _ in range(n)]
        for i in range(2, n):
            if A[i] + A[i-2] == A[i-1] * 2:
                dp[i] = dp[i-1] + 1
        return sum(dp

File: algorithms/400-499/413. Arithmetic Slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& A) {
        if (A.empty()) return 0;
        int n = A.size();
        if (n < 3) return 0;
        vector<int> dp(n,0);
        for (int i = 2; i < n; ++i){
            if (A[i] + A[i-2] == 2 * A[i-1]){
                dp[i] = dp[i-1] + 1;
            }
        }
        return accumulate(dp.begin(), dp.end(), 0);
        
    }
};